src/solvers/flattening/arrays.h

```diff
@@ -77,6 +82,27 @@ class arrayst:public equalityt
   typedef std::map<std::size_t, index_sett> index_mapt;
   index_mapt index_map;
 
+  enum class constraint_typet
+  {
+    ARRAY_ACKERMANN,
+    ARRAY_EQUALITY,
+    ARRAY_WITH,
+    ARRAY_WITH_OTHER,
+    ARRAY_IF_TRUE,
+    ARRAY_IF_FALSE,
+    ARRAY_OF,
+    ARRAY_TYPECAST,
+    ARRAY_CONSTANT,
+    ARRAY_NON_CONSTANT,
+    ARRAY_COMPREHENSION
+  };
```

Comment:
Is there a clean way of extending the existing `lazy_typet` enum class into a new enum class to include `ARRAY_EQUALITY`? Do not want to change the meaning of the `lazy_typet` by adding `ARRAY_EQUALITY` to the existing enum class.

Reply:
@peterschrammel